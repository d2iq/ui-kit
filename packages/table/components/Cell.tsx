import * as React from "react";
import ReactDOM from "react-dom";
import { cx } from "emotion";
import { innerCellCss, cellAlignmentCss, textCapitalize } from "../style";

export type TextAlign = "left" | "right" | "center";
export interface CellProps {
  textAlign?: TextAlign;
  lowerCase?: boolean;
  children: React.ReactElement<HTMLElement> | string;
  className?: string;
  role?: string;
  onMouseEnter?: (event?: React.SyntheticEvent<HTMLElement>) => void;
  onMouseLeave?: (event?: React.SyntheticEvent<HTMLElement>) => void;
  onFocus?: (event?: React.SyntheticEvent<HTMLElement>) => void;
  onBlur?: (event?: React.SyntheticEvent<HTMLElement>) => void;
}

interface CellState {
  renderedTextContent?: string | null;
}

class Cell extends React.PureComponent<CellProps, CellState> {
  constructor(props) {
    super(props);

    this.state = {
      renderedTextContent: null
    };
  }

  getTextContent() {
    const node = ReactDOM.findDOMNode(this);
    if (node) {
      return node.textContent;
    }
  }

  componentDidMount() {
    this.setState({ renderedTextContent: this.getTextContent() });
  }

  componentDidUpdate() {
    if (this.getTextContent() !== this.state.renderedTextContent) {
      this.setState({ renderedTextContent: this.getTextContent() });
    }
  }

  render() {
    const { children, className, textAlign, lowerCase, ...other } = this.props;

    return (
      <div
        title={this.state.renderedTextContent || undefined}
        className={cx(
          innerCellCss,
          cellAlignmentCss(textAlign || "left"),
          lowerCase ? null : textCapitalize,
          className
        )}
        {...other}
      >
        {children}
      </div>
    );
  }
}

export default Cell;
