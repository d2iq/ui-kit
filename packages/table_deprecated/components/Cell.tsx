import * as React from "react";
import ReactDOM from "react-dom";
import { cx } from "@emotion/css";
import { innerCellCss, cellAlignmentCss, textCapitalize } from "../style";

export type TextAlign = "left" | "right" | "center";
export interface CellProps extends React.HTMLAttributes<HTMLDivElement> {
  textAlign?: TextAlign;
  capitalize?: boolean;
  children: React.ReactElement<HTMLElement> | string;
  className?: string;
}

interface CellState {
  renderedTextContent?: string | null;
}

class Cell extends React.PureComponent<CellProps, CellState> {
  state = {
    renderedTextContent: null
  };

  getTextContent() {
    return ReactDOM.findDOMNode(this)?.textContent;
  }

  componentDidMount() {
    this.setState({ renderedTextContent: this.getTextContent() });
  }

  componentDidUpdate() {
    if (this.getTextContent() !== this.state.renderedTextContent) {
      this.setState({ renderedTextContent: this.getTextContent() });
    }
  }

  render() {
    const { children, className, textAlign, capitalize, ...other } = this.props;

    return (
      <div
        title={this.state.renderedTextContent || undefined}
        className={cx(
          innerCellCss,
          cellAlignmentCss(textAlign || "left"),
          { [textCapitalize]: capitalize },
          className
        )}
        {...other}
      >
        {children}
      </div>
    );
  }
}

export default Cell;
