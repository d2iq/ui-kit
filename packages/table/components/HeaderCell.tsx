import React from "react";

import Cell, { CellProps } from "./Cell";
import styled from "react-emotion";
import { headerCellCss } from "../style";
import { textTruncate } from "../../shared/styles/styleUtils";

// Make capitalize false only if it is explicitly passed as false.
// If it is undefned, make it true.
const HeaderCell = (props: CellProps) => (
  <Cell capitalize={props.capitalize === false ? false : true} {...props} />
);

export default styled(HeaderCell)`
  ${headerCellCss};
  ${textTruncate};
`;
