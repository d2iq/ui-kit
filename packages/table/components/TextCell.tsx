import { default as Cell } from "./Cell";
import styled from "react-emotion";

export default styled(Cell)`
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`;
