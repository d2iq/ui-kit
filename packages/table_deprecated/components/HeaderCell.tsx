import React from "react";
import { cx } from "@emotion/css";
import { useId } from "react-id-generator";

import Cell, { CellProps } from "./Cell";
import { headerCellCss, headerCellInner } from "../style";
import {
  textTruncate,
  display,
  flexItem,
  padding
} from "../../shared/styles/styleUtils";
import { Icon } from "../../icon";
import { SystemIcons } from "../../icons/dist/system-icons-enum";
import { greyLightDarken2 } from "../../design-tokens/build/js/designTokens";
import { Tooltip } from "../../tooltip";

export interface HeaderCellProps extends CellProps {
  tooltipContent?: React.ReactNode;
  tooltipId?: string;
}

const HeaderCell = ({
  capitalize = true,
  children,
  tooltipContent,
  tooltipId = useId(1, "columnHeaderTooltip-")[0],
  className,
  ...other
}: HeaderCellProps) => (
  <Cell
    capitalize={capitalize}
    className={cx(className, headerCellCss)}
    {...other}
  >
    <div className={cx(display("inline-flex"), headerCellInner)}>
      <div
        className={cx(textTruncate, flexItem("grow"), padding("right", "xxs"))}
      >
        {children}
      </div>
      {tooltipContent ? (
        <Tooltip
          id={tooltipId}
          trigger={
            <Icon
              shape={SystemIcons.CircleQuestion}
              size="xs"
              color={greyLightDarken2}
            />
          }
        >
          {tooltipContent}
        </Tooltip>
      ) : null}
    </div>
  </Cell>
);

export default HeaderCell;
