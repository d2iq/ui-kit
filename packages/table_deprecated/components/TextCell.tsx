import * as React from "react";
import { default as Cell, CellProps } from "./Cell";
import { textTruncate } from "../../shared/styles/styleUtils";
import { cx } from "@emotion/css";

const TextCell = ({ className, ...other }: CellProps) => (
  <Cell className={cx(className, textTruncate)} {...other} />
);

export default TextCell;
