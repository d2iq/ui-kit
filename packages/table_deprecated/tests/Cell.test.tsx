import * as React from "react";
import { createSerializer } from "@emotion/jest";
import { render } from "enzyme";
import toJson from "enzyme-to-json";

import Cell from "../components/Cell";
import HeaderCell from "../components/HeaderCell";
import TextCell from "../components/TextCell";

expect.addSnapshotSerializer(createSerializer());

describe("Cell", () => {
  it("renders correctly", () => {
    expect(toJson(render(<Cell>This is content</Cell>))).toMatchSnapshot();
  });
});

describe("HeaderCell", () => {
  it("renders correctly", () => {
    expect(
      toJson(render(<HeaderCell>This is content</HeaderCell>))
    ).toMatchSnapshot();
  });
});

describe("TextCell", () => {
  it("renders correctly", () => {
    expect(
      toJson(render(<TextCell>This is content</TextCell>))
    ).toMatchSnapshot();
  });
});
