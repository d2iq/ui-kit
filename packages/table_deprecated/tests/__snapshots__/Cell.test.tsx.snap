// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell renders correctly 1`] = `
.emotion-0 {
  height: auto;
  box-sizing: border-box;
  padding: 7px;
  z-index: 2;
  text-align: left;
  padding-right: ;
}

<div
  class="emotion-0"
>
  This is content
</div>
`;

exports[`HeaderCell renders correctly 1`] = `
.emotion-0 {
  height: auto;
  box-sizing: border-box;
  padding: 7px;
  z-index: 2;
  text-align: left;
  padding-right: ;
  text-transform: capitalize;
  font-weight: bold;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
  position: relative;
  z-index: 2;
}

.emotion-2 {
  overflow: hidden;
  overflow: -moz-hidden-unscrollable;
  text-overflow: ellipsis;
  white-space: nowrap;
  box-sizing: border-box;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
  width: auto;
  padding-right: 4px;
}

<div
  class="emotion-0"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2"
    >
      This is content
    </div>
  </div>
</div>
`;

exports[`TextCell renders correctly 1`] = `
.emotion-0 {
  height: auto;
  box-sizing: border-box;
  padding: 7px;
  z-index: 2;
  text-align: left;
  padding-right: ;
  overflow: hidden;
  overflow: -moz-hidden-unscrollable;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<div
  class="emotion-0"
>
  This is content
</div>
`;
